c10/util/env.cpp

```diff
@@ -0,0 +1,68 @@
+#include <c10/util/Exception.h>
+#include <c10/util/env.h>
+#include <cstdlib>
+#include <shared_mutex>
+
+namespace c10::utils {
+
+static std::shared_mutex env_mutex;
+
+// Checks an environment variable is set.
+bool has_env(const char* name) {
+  std::shared_lock lk(env_mutex);
```

Comment:
I'm confused why this is a shared_lock and not a unique_lock for all functions?

Reply:
Yes, I haven't added a putenv/setenv yet.